File: jpress-engine/src/main/java/by/ibragimov/jpress/generator/template/TemplateEngine.java
package by.ibragimov.jpress.generator.template;

import java.util.Map;

/**
 * Interface that all template engines used in jpress should follow.
 *
 * @author Ibragimov Ruslan
 */
public interface TemplateEngine {

    /**
     * Render particular page using provided data and template.
     *
     * @param template path in classpath to template with theme prefix. example: jfifteen/404.jade
     * @param model object that used to fill template
     * @return fully rendered page
     */
    String render(final String template, final Map<String, Object> model);
}

File: kpress-engine/src/main/kotlin/by/heap/kpress/generator/template/TemplateEngine.kt
package by.heap.kpress.generator.template

/**
 * Interface that all template engines used in kpress should follow.
 *
 * @author Ruslan Ibragimov
 */
interface TemplateEngine {

    /**
     * Render particular page using provided data and template.
     *
     * @param template path in classpath to template with theme prefix. example: kfifteen/404.jade
     * @param model object that used to fill template
     * @return fully rendered page
     */
    fun render(template: String, model: MutableMap<String, Any>): String
}
